pkgs/development/compilers/rust/make-rust-platform.nix

```
@@ -7,22 +7,59 @@
 , ...
 }:
 
-rec {
-  rust = {
-    rustc = lib.warn "rustPlatform.rust.rustc is deprecated. Use rustc instead." rustc;
-    cargo = lib.warn "rustPlatform.rust.cargo is deprecated. Use cargo instead." cargo;
-  };
-
+let
   fetchCargoTarball = buildPackages.callPackage ../../../build-support/rust/fetch-cargo-tarball {
     git = buildPackages.gitMinimal;
     inherit cargo;
   };
 
-  buildRustPackage = callPackage ../../../build-support/rust/build-rust-package {
+  # Derivations built with `buildRustPackage` can already be overridden with
+  # `override`, `overrideAttrs`, and `overrideDerivation`.
+  # This function introduces `overrideRust` and it overrides the call to
+  # `buildRustPackage`.
```

Some motivation would be nice.
```suggestion
  # `buildRustPackage`. This is useful for ...
```